app/webpacker/components/BookmarkIcon/index.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { Icon, Popup } from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import WCAQueryClientProvider from '../../lib/providers/WCAQueryClientProvider';
+import { bookmarkCompetition, unbookmarkCompetition } from './api/bookmarkCompetition';
+import I18n from '../../lib/i18n';
+
+export default function Wrapper({ competitionIsBookmarked, competitionId }) {
+  return (
+    <WCAQueryClientProvider>
+      <BookmarkIcon
+        competitionInitiallyBookmarked={competitionIsBookmarked}
+        competitionId={competitionId}
+      />
+    </WCAQueryClientProvider>
+  );
+}
+
+function BookmarkIcon({ competitionInitiallyBookmarked, competitionId }) {
+  const [isBookmarked, setIsBookmarked] = useState(competitionInitiallyBookmarked);
+
+  const { mutate: bookmarkCompetitionMutation, isPending: isBookmarking } = useMutation({
+    mutationFn: bookmarkCompetition,
+    onSuccess: () => {
+      setIsBookmarked(!isBookmarked);
```

Comment:
It's safer to explicitly set it, just in case the state somehow gets out of sync. It might be possible for the user to click again while the initial mutation is processing or something, which might flip it twice? (Might be worth disabling the icon while it's loading?)
```suggestion
      setIsBookmarked(true);
```
Likewise below, but with false.

Also, could be simplified to one line:
```
onSuccess: () => setIsBookmarked(true),
```